rio/subsystems/drivetrain.py

```diff
@@ -105,68 +120,141 @@ def configureNetworkTables(self):
+            self.constants["pose_x"],
+            self.constants["pose_y"],
         )
 
+        # Reset both motor controllers to factory defaults
+        self.drive_motor.restoreFactoryDefaults()
+        self.steer_motor.restoreFactoryDefaults()
+
+        # Construct the PID controllers
+        self.steer_PID = self.steer_motor.getPIDController()
+
+        # Assign PID Values
+        # TODO: Drive motor too
+        self.steer_PID.setD(self.pid["steer"]["kd"])
+        self.steer_PID.setI(self.pid["steer"]["ki"])
+        self.steer_PID.setP(self.pid["steer"]["kp"])
+        # self.steer_PID.setFF(1)
+        self.steer_PID.setIMaxAccum(self.pid["steer"]["maxi"])
+        self.steer_PID.setOutputRange(
+            self.pid["steer"]["min_power"],
+            self.pid["steer"]["max_power"],
+        )
+
+        # Assign ratios
+        # TODO: Drive motor too
+        # self.steer_motor_encoder.setPositionConversionFactor()
+
+        # Save all settings to flash
+        # TODO: Is it a good idea to do this **every** reboot?
+        self.drive_motor.burnFlash()
+        self.steer_motor.burnFlash()
+
+        # Other sensors
+        self.steer_motor_encoder = self.steer_motor.getEncoder()
+
         # Current state variables
-        self.is_zeroed = False
-        self.state = kinematics.SwerveModuleState(
-            0, geometry.Rotation2d(0)
-        )  # This module state is default 0 speed, and 0 rotation
+        self.isZeroed = False
+        # By default: 0 speed, and 0 rotation
+        self.desiredState = kinematics.SwerveModuleState(0, geometry.Rotation2d(0))
 
-    def getTranslation(self):
+    def getPose(self):
         return self.module_pose
 
-    def setModuleState(self, newState):
+    def setDesiredState(self, newState):
         """
-        Important method that updates
-        the "state" (steering and speed)
-        of a module.
+        Updates the current desired state,
+        where we want this module to now point.
         """
 
-        # TODO: Use optimization at some point
-        self.state = newState
+        def __init__(self):
+
+            # Optimize the input command to reduce unneeded motion.
+            optimizedState = kinematics.SwerveModuleState.optimize(
+                newState, self.getState().angle
+            )
+
+            # Get the desired position (in neo rotations) given by the optimized module state
+            currentRef = (optimizedState.angle.radians() / (2 * math.pi)) * self.pid[
+                "steer"
+            ]["ratio"]
+
+            # this checks if the desiredState is more than 349 and currentRef is 59 then adds 2 pi to currentRef
+            if (
+                self.desiredState >= currentRef
+                and self.desiredState.angle.radians() >= 6.3490912
+                and currentRef.angle.radians() <= 1.0297
+            ):
+                currentRef = currentRef.angle.radians() + math.pi + math.pi
```

Comment:
This is an odd way of adding 2pi but i see what you're trying to do

Reply:
```python
2 * math.pi```